src/Model/SourcesRepository.php

```diff
@@ -79,20 +79,22 @@ public function getByOrderId(string $orderId): SourcesInterface
      * @param string $orderId
      * @param string $itemSku
      *
-     * @return SourceSelectionItem
+     * @return SourceSelectionItem []
      * @throws NoSuchEntityException
      */
-    public function getSourceItemBySku(string $orderId, string $itemSku): SourceSelectionItem
+    public function getSourceItemBySku(string $orderId, string $itemSku): array
     {
         $sourceSelectionItems = $this->sourcesConverter->convertSourcesJsonToSourceSelectionItems(
             $this->getByOrderId($orderId)->getSources()
         );
 
-        if (!array_key_exists($itemSku, $sourceSelectionItems)) {
-            throw new NoSuchEntityException(__('Source model with the sku "%1" does not exist', $itemSku));
+        foreach ($sourceSelectionItems as $sourceSelectionItem) {
+            if ($sourceSelectionItem->getSku() !== $itemSku) {
+                $sourceSelectionItem->setSourceCode('default');
+            }
         }
 
-        return $sourceSelectionItems[$itemSku];
+        return $sourceSelectionItems;
```

Comment:
The repository method suggests that only the source items for the matching SKU are returned. I would expect something like this:

```php
    public function getSourceItemBySku(string $orderId, string $itemSku): array
    {
        $sourceSelectionItems = $this->sourcesConverter->convertSourcesJsonToSourceSelectionItems(
            $this->getByOrderId($orderId)->getSources()
        );

        $items = [];
        foreach ($sourceSelectionItems as $sourceSelectionItem) {
            if ($sourceSelectionItem->getSku() === $itemSku) {
                $items[] = $sourceSelectionItem;
            }
        }

        return $items;
    }
```